File: mbv_open_api_sdk_csharp/mbv_open_api_sdk_csharp/com/mbv/open/api/sdk/MbClient.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using com.mbv.open.api.sdk.request;
using com.mbv.open.api.sdk.response;

namespace com.mbv.open.api.sdk
{
    public interface MbClient
    {
        T Execute<T>(BaseRequest<T> request) where T : BaseResponse;
    }
}

File: mbv_open_api_sdk/src/main/java/com/mbv/open/api/sdk/MbClient.java
package com.mbv.open.api.sdk;

import com.mbv.open.api.sdk.exception.SdkException;
import com.mbv.open.api.sdk.request.BaseRequest;
import com.mbv.open.api.sdk.response.BaseResponse;

public abstract interface MbClient
{
  public abstract <T extends BaseResponse> T execute(BaseRequest<T> request)
    throws SdkException;
} 
